src/dmd/cppmangle.d

```diff
@@ -1259,58 +1261,45 @@ extern(C++):
          * Ds       char16_t
          * u <source-name>  # vendor extended type
          */
-        char c;
-        char p = 0;
+
+        // Handle any target-specific basic types.
+        if (auto tm = target.cppTypeMangle(t))
+        {
+            bool builtin = (t.ty == Tvoid || t.ty == Tbool || t.isintegral() || t.isreal());
+            writeBasicType(t, tm.toDString(), builtin);
```

Comment:
If you need to do a strlen, I suggest having target.cppTypeMangle() return a string in the first place.

Reply:
The function is implemented in c++, which doesn't know how D strings are passed around.

Can just drop the `[]` from writeBasicType instead.

Reply:
(  Reversing the order (first check target.cppTypeMangle before doing the hardcoded switch) is helpful: backends no longer need to modify dmdfe source to override the hardcoded mangles. Nice.  )